xocto/ranges.py

```diff
@@ -901,12 +901,12 @@ def __and__(
     ) -> Optional["FiniteDatetimeRange"]:
         return self.intersection(other)
 
-    @property
-    def days(self) -> int:
+    def days(self, tz: datetime.tzinfo) -> int:
         """
         Return the number of days between the start and end of the range.
         """
-        return (self.end - self.start).days
+        range_ = self.localize(tz)
```

Comment:
❓ How about disallowing differing zones, just for this property?

Requiring `tz` means we're going to need to update all call sites anyway, so it's easy enough to do `period.localize(tz).days` instead of `period.days(tz)`. That keeps this property simple and makes the caller a little more explicit with its intent.